src/utils.ts

```diff
@@ -178,8 +194,8 @@ export function areMapsEqual(
         isValueEqual = false;
 
         b.forEach((bValue, bKey) => {
-          if (!isValueEqual && isEqual(aKey, bKey, meta)) {
-            isValueEqual = isEqual(aValue, bValue, meta);
+          if (!isValueEqual && isEqual(aKey, bKey, undefined, a, b, meta)) {
```

Comment:
I mention it in more detail below related to `Set` comparisons, but `key` being `undefined` here makes sense from an implementation perspective, but may not from a consumption perspective. The idea I had there related to an "iteration index" may be applicable here as well.

Reply:
Here, I think we shouldn't use a custom comparator and pass these values. This comparison only makes sense internally to make sure we compare the right two Map values. When I wrote the test for the new params I forgot this case at first, and I think users of the library don't know keys from Maps will show up as values in the custom comparator when using `createCustomEqual`. It could even break the comparison for some extreme cases.

From the specs for Map:

> Distinct key values are discriminated using the SameValueZero comparison algorithm.

So we could use sameValueZero comparison, or use `.has` and `.get` instead of the second loop. (This is the approach taken by fast-deep-equal). 
Or we could keep deep comparison of keys but use the basic `compare`.
What do you think ?

Reply:
The reason for this internal comparison is to allow value-based comparisons. Your reference to the spec is correct, however an architectural decision was "deep equality is not exclusive to values". Example:

```ts
const mapA = new Map([[{ foo: 'bar'}, { bar: 'baz' }]]);
const mapB = new Map([[{ foo: 'bar'}, { bar: 'baz' }]]);
```

If we used SMZ for comparison, these would never be equal, even though the map contents (key + value) are identical.

In regards to the second point of using the basic `compare` instead of the provided `isEqual`, I went back and forth on this but eventually came to the decision that creating different comparisons for key vs value left a gap in capabilities. Creating a custom equality comparator is a rare use-case, so instead of limit functionality (what if a custom `key` comparison was desired?) I decided to leave the implementation details of the comparison to the consumer. I try to follow a general guideline of "common use-cases are simple, advanced use-cases are understandable, extreme edge-cases are possible".

Reply:
Ok, I understand, it makes sense. We could maybe add a line about it in the docs ?

Reply:
Good call, it definitely should be mentioned as it is not obvious. Prior to publish of this change I'll update the docs to provide that nuance.